matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/util/database/RealmMigrator.kt

```diff
@@ -18,10 +18,13 @@ package org.matrix.android.sdk.internal.util.database
 
 import io.realm.DynamicRealm
 import io.realm.RealmObjectSchema
+import org.matrix.android.sdk.internal.util.database.exceptions.RealmSchemaVersionException
 import timber.log.Timber
 
-abstract class RealmMigrator(private val realm: DynamicRealm,
-                             private val targetSchemaVersion: Int) {
+abstract class RealmMigrator(private val realm: DynamicRealm) {
+
+    private val targetSchemaVersion = calculateSchemaVersion(this::class.java.simpleName)
```

Comment:
this will break if any forks have R8/proguard class name obfuscating enabled, might be worth adding a proguard rule to be 100% sure the names are kept (if we want to use reflection)

Reply:
Yes, or we can skip it completely its fine by me.